jupyter-lib/api/src/main/kotlin/org/jetbrains/kotlinx/jupyter/api/fieldsHandling.kt

```diff
@@ -36,3 +37,25 @@ class FieldHandlerByClass(
 ) : FieldHandler {
     override fun acceptsType(type: KType) = type.isSubtypeOf(kClass.starProjectedType)
 }
+
+data class VariableDeclaration(
+    val name: VariableName,
+    val value: Any?,
+    val type: KType,
+    val isMutable: Boolean = false,
+) {
+    constructor(
+        name: VariableName,
+        value: Any?,
+        isMutable: Boolean = false
+    ) : this(
+        name,
+        value,
+        value?.let { it::class.starProjectedType } ?: Any::class.createType(nullable = true),
+        isMutable
+    )
+}
+
+fun KotlinKernelHost.declareVariable(variable: VariableDeclaration) = declareVariables(listOf(variable))
+fun KotlinKernelHost.declareVariable(name: VariableName, value: Any?) = declareVariable(VariableDeclaration(name, value))
+fun KotlinKernelHost.declareVariables(variables: Map<VariableName, Any?>) = declareVariables(variables.map { VariableDeclaration(it.key, it.value) })
```

Comment:
Probably better to use `vararg pairs: Pair<VariableName, Any?>`. It would cover both this case and the line before.